### PR TITLE
Relax IP restrictions to allow hostnames, domain names and TLDs

### PR DESCRIPTION
**Pull request by exalented**
- Allow user to type in letters, hyphens and underscores
- Remove validation function

### CODE REVIEW

File: app/ui/launcher/add.dialog.c [hunk 1]
```diff
@@ -103,7 +101,7 @@ static void dialog_cb(lv_event_t *event) {
     uint16_t btn = lv_msgbox_get_active_btn(dialog);
     if (btn == 1) {
         const char *address = lv_textarea_get_text(controller->input);
-        if (!validate_ip4(address))return;
+        if (!address)return;
```
Comment (mariotaku) [inline app/ui/launcher/add.dialog.c:104]: Use `strlen`

File: app/ui/launcher/add.dialog.c [hunk 2]
```diff
@@ -67,7 +65,7 @@ static lv_obj_t *create_dialog(lv_obj_controller_t *self, lv_obj_t *parent) {
     lv_obj_set_grid_cell(ip_input, LV_GRID_ALIGN_STRETCH, 0, 3, LV_GRID_ALIGN_STRETCH, 1, 1);
     lv_textarea_set_placeholder_text(ip_input, "IPv4 address only");
     lv_textarea_set_one_line(ip_input, true);
-    lv_textarea_set_accepted_chars(ip_input, "0123456789.");
+    lv_textarea_set_accepted_chars(ip_input, ".0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz");
```
Comment (mariotaku) [inline app/ui/launcher/add.dialog.c:68]: Domain name might contain - and _

**Review by mariotaku — CHANGES_REQUESTED** [review]
Great work! I have left some review comments, please take a look.

File: app/ui/launcher/add.dialog.c (hunk 1 above)
Comment (exalented) [reply app/ui/launcher/add.dialog.c]: We talked about only checking the first character in the string

**Review by mariotaku — APPROVED** [review]

File: app/ui/launcher/add.dialog.c [hunk 3]
```diff
@@ -103,7 +101,7 @@ static void dialog_cb(lv_event_t *event) {
     uint16_t btn = lv_msgbox_get_active_btn(dialog);
     if (btn == 1) {
         const char *address = lv_textarea_get_text(controller->input);
-        if (!validate_ip4(address))return;
+        if (!address && !address[0])return;
```
Comment (mariotaku) [inline app/ui/launcher/add.dialog.c:104]: `if (!address || !address[0])return;`

File: app/ui/launcher/add.dialog.c [hunk 4]
```diff
@@ -117,7 +115,7 @@ static void dialog_cb(lv_event_t *event) {
 static void input_changed_cb(lv_event_t *event) {
     add_dialog_controller_t *controller = lv_event_get_user_data(event);
     const char *address = lv_textarea_get_text(controller->input);
-    if (validate_ip4(address)) {
+    if (address) {
```
Comment (mariotaku) [inline app/ui/launcher/add.dialog.c:118]: `if (address && address[0]) {`

**Review by mariotaku — CHANGES_REQUESTED** [review]

File: app/ui/launcher/add.dialog.c (hunk 3 above)
Comment (exalented) [reply app/ui/launcher/add.dialog.c:104]: right

**Review by mariotaku — APPROVED** [review]

**Pull request merged**